Fibank.rb

```diff
@@ -0,0 +1,128 @@
+require 'watir'
+require 'date'
+require 'nokogiri'
+
+require_relative 'account'
+require_relative 'transaction'
+
+class Fibank
+  MAIN_URL = 'https://my.fibank.bg/oauth2-server/login?client_id=E_BANK'
+
+  def initialize
+    @browser = Watir::Browser.start(MAIN_URL)
+    @accounts = []
+  end
+
+  def execute
+    connect
+    collect_accounts
+    collect_transactions
+    print_result
+  end
+
+  private
+
+  def connect
+    @browser.link(id: 'demo-link').click
+    sleep 2
+  end
+
+  def collect_accounts
+    html = fetch_accounts
+
+    parse_accounts(html)
+  end
+
+  def collect_transactions
+    @accounts.each do |account|
+      html = fetch_transactions(account)
+
+      parse_transactions(account, html)
+    end
+  end
+
+  def fetch_accounts
+    Nokogiri.parse(@browser.html).css('table#dashboardAccounts tbody tr')
+  end
+
+  def fetch_transactions(account)
+    go_home
+
+    sleep 2
+
+    select_account(account)
+
+    Nokogiri.parse(@browser.html)
+  end
+
+  def parse_accounts(html)
+    html.each do |tr|
+      name = tr.xpath('.//p[@bo-bind="row.acDesc"]').text
+      currency = tr.xpath('.//span[@bo-bind="row.ccy"]').text
+      balance = tr.xpath('.//span[@bo-bind="row.acyAvlBal | sgCurrency"]') \
```

Comment:
```ruby
tr.css("span[bo-bind='acyAvlBal']")

Reply:
Done

Reply:
You have method `to_amount` you can use it for `balance`

Reply:
Ok